add code to compute NH pressure at interfaces

mt5555 opened the pull request:
add an interface to the dycore's nonhydrostatic pressure on layer interfaces

based on the analysis here:
https://acme-climate.atlassian.net/wiki/spaces/NGDNA/pages/2579530246/NH+surface+pressure

we do not change the current approach at the surface (using the mu=1 approximation), but this does switch from a cell average type algorithm to more accurate interpolation for the interior interface values.

[BFB] since this is a diagnostic not currently used

mt5555 commented:
@bartgol : does this require corresponding changes in HOMMEXX?

bartgol commented:
This is just to couple to eam, right? pnh_i is never _used_ inside Homme, so there should be no need for changes in Hommexx.

rljacob commented:
Is the title right?  "compute to compute" ?

mt5555 commented:
replaced algorithm.  Removed use of weights based on eta coordinate (per  @oksanaguba recommendation) and replaced with linear interpolation in the hydrostatic pressure coordinate.  

This has the advantage that if pnh=pi (on midpionts), then the interpolation pnh_i = pi_i.

This is a desirable property, since in the NH model, pnh is often very close to pi, and so pnh_i should be close to pi_i.

bartgol commented on components/homme/src/theta-l/share/eos.F90:
```diff
@@ -210,16 +210,12 @@ subroutine pnh_and_exner_from_eos2(hvcoord,vtheta_dp,dp3d,dphi,pnh,exner,&
    if (present(pnh_i_out)) then
       ! boundary values already computed. interpolate interior
+      ! use linear interpolation in hydrostatic pressure coordinate
+      ! if pnh=pi, then pnh_i will recover pi_i
```
I don't think this statement is true, since linear interpolation is not the inverse of midpoint averaging.

mt5555 replied:
it's true because dp3d=delta pi.
